setup_trinity.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
```

Comment:
I would feel better if each package had it's own `setup.py` (without any suffix) directly in a package sublevel directory. Makes packages more self-contained. Essentially something like this.

```
evm/
├── src/
│   ├── something.py
│   ├── ...
├── setup.py
├── README.md
p2p/
├── src/
│   ├── something.py
│   ├── ...
├── setup.py
├── README.md
trinity/
├── src/
│   ├── something.py
│   ├── ...
├── setup.py
├── README.md
README.md
tox.ini
...
```

This is the defacto standard in the JS/TS camp but I'm not sure if there is anything that makes this impractical in the Python ecosystem?

Reply:
I'm avoiding doing this, largely because I don't want to move the `evm` stuff, then move it back later, so I'm thinking of this as a short term (few months maybe) stop gap solution.

Reply:
*I know, I know*, while I'm looking at it from a *monorepos are so much easier to maintain* kind of perspective :wink: 

But I realize I may be alone in this camp so I'm fine to go with the majority vote.

Reply:
Your camp may very well be lonely 😄 

I've never tried the approach but I don't think I want to at this time.  Also, with github it seems like a CI and permissions quagmire.

Reply:
>Also, with github it seems like a CI and permissions quagmire

Yep, point taken. It's just much easier to handle PRs that touch different parts of the system, handle labels, manage releases and versions etc

Anyway, there are pros and cons for both approaches and I also have to admit that I have zero experience regarding Python and monorepos and what kind of special problems there may be.

I can happily party with the other camp, too ;)